components/script/dom/node.rs

```diff
@@ -118,6 +127,9 @@ pub struct Node {
     /// The document that this node belongs to.
     owner_doc: MutNullableDom<Document>,
 
+    /// The shadow root this node belongs to.
```

Comment:
Could you document here what does this return for a `ShadowRoot` itself? Does it return `self`? Or something else?